### PR TITLE
Add problems solved rating history graph

### LINKED ISSUE #2: plot average problem solved rating over time along with scatterplot

### PR DESCRIPTION
**Pull request by aryamanarora**
Resolves #2.

Some issue that geniosities need to fix:
* I'm still using the old api system before it was abstracted to another file
* there's basically no error handling
* pick a better name (the command is called `geniosity` right now)

Don't merge it until that's fixed or the bot will die.

### CODE REVIEW

File: tle/cogs/codeforces.py [hunk 1]
```diff
@@ -249,6 +249,93 @@ def __init__(self, bot):
...
+        for submission in submissions:
+            if submission['verdict'] == 'OK':
+                problem = submission['problem']
+                # CF problems don't have IDs! Just hope (name, rating) pairs don't clash?
+                name = problem['name']
+                rating = problem.get('rating')
+                t = submission['author']['participantType']
+                startTimeSeconds = submission.get('creationTimeSeconds')
+                if rating:
+                    key = 2
```
Comment (cheran-senthil) [inline tle/cogs/codeforces.py]: Use a single line if else

File: tle/cogs/codeforces.py [hunk 2]
```diff
@@ -249,6 +249,93 @@ def __init__(self, bot):
...
+        ratings = [[], [], []]
+        ts, rs = [], []
+        rkey = ['PRACTICE', 'VIRTUAL', 'CONTESTANT']
+        for submission in submissions:
+            if submission['verdict'] == 'OK':
+                problem = submission['problem']
+                # CF problems don't have IDs! Just hope (name, rating) pairs don't clash?
+                name = problem['name']
+                rating = problem.get('rating')
+                t = submission['author']['participantType']
```
Comment (cheran-senthil) [inline tle/cogs/codeforces.py]: Check meooow's refactoring using named tuples.

File: tle/cogs/codeforces.py [hunk 3]
```diff
@@ -249,6 +249,93 @@ def __init__(self, bot):
...
+            plt.axvspan(loc, loc, facecolor='white')
+        
+        # moving average for loop cancer
+        if len(rs) > bin_size:
+            avg_ts = []
+            avg_rs = []
+            cur_t, cur_r = 0, 0
+            for i in range(bin_size - 1):
+                cur_t += datetime.datetime.timestamp(ts[i])
+                cur_r += rs[i]
```
Comment (cheran-senthil) [inline tle/cogs/codeforces.py]: Use sum, ideally merge with above loop for simplicity

File: tle/cogs/codeforces.py [hunk 4]
```diff
@@ -249,6 +249,93 @@ def __init__(self, bot):
...
+            for i in range(bin_size - 1, len(rs)):
+                cur_t += datetime.datetime.timestamp(ts[i])
+                cur_r += rs[i]
+                avg_ts.append(datetime.datetime.fromtimestamp(cur_t / bin_size))
+                avg_rs.append(cur_r / bin_size)
+                cur_t -= datetime.datetime.timestamp(ts[i - bin_size + 1])
+                cur_r -= rs[i - bin_size + 1]
```
Comment (cheran-senthil) [inline tle/cogs/codeforces.py]: Use a list comprehension and compress this.

**Comment by krofna** [comment]
Resolved.

**Pull request closed without merging**